raphtory/src/search/edge_index.rs

```diff
@@ -0,0 +1,347 @@
+        let mut const_writers = initialize_edge_const_property_indexes(
+            graph,
+            edge_index.constant_property_indexes.clone(),
+            const_property_keys,
+        )?;
+
+        let temporal_property_keys = graph
+            .edge_meta()
+            .temporal_prop_meta()
+            .get_keys()
+            .into_iter();
+        let mut temporal_writers = initialize_edge_temporal_property_indexes(
+            graph,
+            edge_index.temporal_property_indexes.clone(),
+            temporal_property_keys,
+        )?;
+
+        let mut writer = edge_index.index.writer(100_000_000)?;
+        let locked_g = graph.core_graph();
+        locked_g.edges_par(&graph).try_for_each(|e_ref| {
+            {
+                let e_view = EdgeView::new(graph.clone(), e_ref);
+                // manufacture &[(usize, Prop)]
+                edge_index.index_edge(graph, e_view, &writer, &const_writers, &temporal_writers)?;
+            }
+            Ok::<(), TantivyError>(())
+        })?;
+
+        // Commit writers
+        for writer_option in &mut const_writers {
+            if let Some(const_writer) = writer_option {
+                const_writer.commit()?;
+            }
+        }
+        for writer_option in &mut temporal_writers {
+            if let Some(temporal_writer) = writer_option {
+                temporal_writer.commit()?;
+            }
+        }
+        writer.commit()?;
+
+        // Reload readers
+        {
+            let const_indexes = edge_index.constant_property_indexes.read()?;
+            for property_index_option in const_indexes.iter().flatten() {
+                property_index_option.reader.reload()?;
+            }
+        }
+        {
+            let temporal_indexes = edge_index.temporal_property_indexes.read()?;
+            for property_index_option in temporal_indexes.iter().flatten() {
+                property_index_option.reader.reload()?;
+            }
+        }
+        edge_index.reader.reload()?;
+
+        Ok(edge_index)
+    }
+
+    pub(crate) fn add_edge_update(
+        &self,
+        graph: &GraphStorage,
+        edge_id: EID,
+        t: TimeIndexEntry,
+        src: VID,
+        dst: VID,
+        layer: usize,
+        props: &[(usize, Prop)]
+    ) -> Result<(), GraphError> {
+        let edge = graph
+            .edge(src, dst)
+            .expect("Edge for internal id should exist.")
+            .at(t.t());
+
+        let const_property_keys = edge.const_prop_keys();
+        let const_writers = initialize_edge_const_property_indexes(
+            graph,
+            self.constant_property_indexes.clone(),
+            const_property_keys,
+        )?;
+
+        let temporal_property_keys = edge.temporal_prop_keys();
+        let temporal_writers = initialize_edge_temporal_property_indexes(
+            graph,
+            self.temporal_property_indexes.clone(),
+            temporal_property_keys,
+        )?;
```

Comment:
shouldn't need to do this here every time, you can handle new property additions in the resolve step